lib/AST/Decl.cpp

```diff
@@ -2977,14 +2976,14 @@ bool AbstractStorageDecl::isFormallyResilient() const {
                             /*treatUsableFromInlineAsPublic=*/true).isPublicOrPackage())
     return false;
 
-  return true;
-}
-
-bool AbstractStorageDecl::isResilient() const {
-  if (!isFormallyResilient())
+  if (!getModuleContext()->isResilient())
     return false;
 
-  return getModuleContext()->isResilient();
+  // Allows bypassing resilience checks for package decls
+  // at use site within a package if opted in, whether the
+  // loaded module was built resiliently or not.
+  return !getDeclContext()->bypassResilienceInPackage(getFormalAccessScope(/*useDC=*/nullptr,
+                                                                           /*treatUsableFromInlineAsPublic=*/true).isPackage());
 }
 
 bool AbstractStorageDecl::isResilient(ModuleDecl *M,
```

Comment:
Actually I think the bypassResilienceInPackage() have to go into the overloads of isResilient() that take a module, because you should be comparing that module's package name against this decl's:

```
if (M->getPackageName() == getModuleContext()->getPackageName()) { ... }
```

Reply:
Updated to compare package names of the decl's defining module `getParentModule()` and the accessing module `getASTContext().MainModule`. The overloaded method that takes in ModuleDecl is only called in a few places with hard coded expansion arguments, and moving bypassing logic into it doesn't seem to cover all relevant cases. Updating call sites to call the overloaded method instead will result in recursively updating their call sites to add a ModuleDecl arg, which will be everywhere. Since `isResilient()` is called by the overloaded method, leaving the logic in it to handle comparison as long as it can get the accessing module seems to make more sense for now. Let me know if there's something I'm missing here.

Reply:
Follow up PR - https://github.com/apple/swift/pull/71663